core/selection.js

```diff
@@ -14,9 +14,21 @@ class Range {
   }
 }
 
+const resolveSelectionRoot = (selectionRoot) => {
+  if (selectionRoot && typeof selectionRoot.getSelection === 'function' && !selectionRoot.isShady) {
+    return selectionRoot;
+  } else if (selectionRoot) {
+    return selectionRoot.ownerDocument || window.document;
+  }
+
+  return window.document;
+}
```

Comment:
The above function should provide a reasonable fallback for use cases where 

* it is assumed that there is one document 
* ShadyDOM is polyfilled in IE